src/targets/gpu/convolution.cpp

```diff
@@ -121,6 +152,69 @@ shape miopen_convolution::find(context& ctx, const shape& output_shape, std::vec
         MIGRAPHX_THROW("MIOpen Convolution: find convolution failed");
     algo = perf.fwd_algo;
 
+    if(enabled(MIGRAPHX_MIOPEN_FIND_2{}))
+    {
+        miopenProblem_t conv_problem;
+        status = miopenCreateConvProblem(&conv_problem, cd.get(), miopenProblemDirectionForward);
+        if(status != miopenStatusSuccess)
+        {
+            MIGRAPHX_THROW("conv problem creation failed");
+        }
+        auto set_tensor_descriptor = [conv_problem](miopenTensorArgumentId_t name,
+                                                    tensor_descriptor& desc) {
+            auto status = miopenSetProblemTensorDescriptor(conv_problem, name, desc.get());
+            if(status != miopenStatusSuccess)
+            {
+                MIGRAPHX_THROW("setting problem tensor description failed");
+            }
+        };
+
+        set_tensor_descriptor(miopenTensorConvolutionX, x_desc);
```

Comment:
This function should be moved to miopen.hpp.

Reply:
Done